symfit/core/minimizers.py

```diff
@@ -358,11 +398,46 @@ class ScipyGradientMinimize(ScipyMinimize, GradientMinimizer):
     """
     Base class for :func:`scipy.optimize.minimize`'s gradient-minimizers.
     """
-    def __init__(self, *args, **kwargs):
-        super(ScipyGradientMinimize, self).__init__(*args, **kwargs)
+    @keywordonly(jacobian=None)
+    def execute(self, **minimize_options):
+        # This method takes the jacobian as an argument because the user may
+        # need to override it in some cases (especially with the trust-constr 
+        # method)
+        jacobian = minimize_options.pop('jacobian')
+        if jacobian is None:
+            jacobian = self.wrapped_jacobian
+        return super(ScipyGradientMinimize, self).execute(jacobian=jacobian, **minimize_options)
+
+    def scipy_constraints(self, constraints):
+        cons = super(ScipyGradientMinimize, self).scipy_constraints(constraints)
+        for con in cons:
+            # FIXME: Just assume that because the objective has a jac, so do
+            # all the constraints
+            con['jac'] = self.resize_jac(con['fun'].eval_jacobian)
+        return cons
+
 
+class ScipyHessianMinimize(ScipyGradientMinimize, HessianMinimizer):
+    """
+    Base class for :func:`scipy.optimize.minimize`'s hessian-minimizers.
+    """
+    @keywordonly(hessian=None)
     def execute(self, **minimize_options):
-        return super(ScipyGradientMinimize, self).execute(jacobian=self.wrapped_jacobian, **minimize_options)
+        # This method takes the hessian as an argument because the user may
+        # need to override it in some cases (especially with the trust-constr 
+        # method)
+        hessian = minimize_options.pop('hessian')
+        if hessian is None:
+            hessian = self.wrapped_hessian
+        return super(ScipyHessianMinimize, self).execute(hessian=hessian, **minimize_options)
+
+    def scipy_constraints(self, constraints):
+        cons = super(ScipyHessianMinimize, self).scipy_constraints(constraints)
+        # FIXME: Just assume that because the objective has a hess, so do
+        # all the constraints
+        for con in cons:
+            con['hess'] = self.resize_hess(con['fun'].eval_hessian)
```

Comment:
I do not think such a key exists: https://docs.scipy.org/doc/scipy-1.1.0/reference/generated/scipy.optimize.minimize.html#scipy.optimize.minimize

Reply:
That is because currently the only Hessian method that can deal with constraints is trust-constr, which coincidentally uses a wildly different constraint format.
For the sake of consistency I think this should be in here, and it might catch some errors where a user tries to do a constraint hessian based minimization using something else than trust-constr.